docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,298 @@
+#################################
+0003 Learning Path Enrollment API
+#################################
+
+******
+Status
+******
+
+**Draft**
+
+.. Standard statuses
+    - **Draft** if the decision is newly proposed and in active discussion
+    - **Provisional** if the decision is still preliminary and in experimental phase
+    - **Accepted** *(date)* once it is agreed upon
+    - **Superseded** *(date)* with a reference to its replacement if a later ADR changes or reverses the decision
+
+    If an ADR has Draft status and the PR is under review, you can either use the intended final status (e.g. Provisional, Accepted, etc.), or you can clarify both the current and intended status using something like the following: "Draft (=> Provisional)". Either of these options is especially useful if the merged status is not intended to be Accepted.
+
+*******
+Context
+*******
+
+Learner enrollments in Learning Paths are captured in the LearningPathEnrollment
+model. In this ADR, it is enhanced with necessary fields and API to perform
+enrollments and store relevant data.
+
+********
+Decision
+********
+
+1. LearningPathEnrollment Model
+===============================
+
+Introduce new fields to the ``LearningPathEnrollment`` model:
+
+#. ``is_active`` as a ``BooleanField`` that will indicate the enrolled/unenrolled
+   state of the learner.
+#. ``history`` as a `django-simple-history`_ ``HistoricalRecords`` field to
+   preserve the record of changes made on an enrollment.
+#. ``enrolled_at`` as a ``DateTimeField`` to record the timestamp of the
+   enrollment. This will be explicitly set/reset when a user is enrolled in the
+   Learning Path. It will hold the latest valid enrollment timestamp in case of
+   un-enrollment and re-enrollment.
+
+.. _django-simple-history: https://django-simple-history.readthedocs.io/en/latest/quick_start.html
+
+2. ManualEnrollmentAudit Model
```

Comment:
Perhaps we could leave this out from the MVP? We may end up extending the `ManualEnrollmentAudit` model after future discussions, so maybe let's add something like:
```rst
.. warning::
   The scope of this model is not finalized, so do not implement it yet.
```

Reply:
Makes sense. We can mark the ADR as provisional and update it once we have the specs finalized.